src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,329 @@
+using LinearAlgebra: eigen, eigvecs
+
+export SymmetricPositiveDefinite, LinearAffineMetric, LogEuclideanMetric
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) =
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n)$, N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = div(N*(N+1), 2)
+
+@doc doc"""
+    LinearAffineMetric <: Metric
+
+The linear affine metric is the metric for symmetric positive definite matrices, that employs
+matrix logarithms and exponentials, which yields a linear and affine metric.
+"""
+struct LinearAffineMetric <: Metric end
+
+@doc doc"""
+    LogEuclideanMetric <: Metric
+
+The LogEuclidean Metric consists of the Euclidean metric applied to all elements after mapping them
+into the Lie Algebra, i.e. performing a matrix logarithm beforehand.
+"""
+struct LogEuclideanMetric <: Metric end
+
+@doc doc"""
+    distance(P,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between `x` and `y`,
+which defaults to the [`LinearAffineMetric`](@ref) induces distance.
+"""
+distance(P::SymmetricPositiveDefinite{N},x,y) where N = distance(MetricManifold(P,LinearAffineMetric()),x,y)
+
+@doc doc"""
+    distance(P,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between `x` and `y`,
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref). The formula reads
+
+```math
+d_{\mathcal P(n)}(x,y) = \lVert \operatorname{Log}(x^{-\frac{1}{2}}yx^{-\frac{1}{2}})\rVert_{\mathrm{F}}.,
+```
+where $\operatorname{Log}$ denotes the matrix logarithm and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the
+matrix Frobenius norm.
+"""
+function distance(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y) where N
+    s = real.( eigen( x,y ).values )
```

Comment:
Using `eigvals` might be faster.

Reply:
Thanks, yes, of course, this is changes in the next commit.